experiments/synthetic.py

```diff
@@ -0,0 +1,234 @@
+"""Module for synthetic experiment."""
+from __future__ import annotations
+
+import copy
+import time
+from collections.abc import Mapping
+from typing import Any, Optional
+
+import hydra
+import numpy as np
+import pandas as pd
+from lifelines.exceptions import ConvergenceError
+from omegaconf.dictconfig import DictConfig
+
+from fedeca.fedeca_core import FedECA
+from fedeca.utils.experiment_utils import effective_sample_size, std_mean_differences
+from fedeca.utils.survival_utils import CoxData
+from fedeca.utils.typing import _SeedType
+
+
+def single_experiment(
+    data_gen: CoxData,
+    n_samples: int,
+    model_configs: Mapping[str, dict | DictConfig],
+    duration_col: str = "time",
+    event_col: str = "event",
+    treated_col: str = "treatment",
+    ps_col: str = "propensity_scores",
+    seed: _SeedType = None,
+    fit_fedeca: Optional[dict[str, Any]] = None,
+    return_propensities: bool = True,
+    return_weights: bool = True,
+):
+    """Perform a single experiment comparing survival models.
+
+    Parameters
+    ----------
+    data_gen : CoxData
+        Data generator instance.
+    n_samples : int
+        Number of samples to generate.
+    model_configs : Mapping[str, dict | DictConfig]
+        Dictionary of configs used to initialize survival models.
+    duration_col : str, optional
+        Column name for event duration, by default "time".
+    event_col : str, optional
+        Column name for event indicator, by default "event".
+    treated_col : str, optional
+        Column name for treatment indicator, by default "treatment".
+    ps_col : str, optional
+        Column name for propensity scores, by default "propensity_scores".
+    seed: {None, int, Sequence[int], SeedSequence, BitGenerator, Generator}, optional
+        The seed for reproducibility. Defaults to None.
+    fit_fedeca: dict[str, Any], optional, by default None
+        Dictionary of kwargs for the fit function of
+        :class:`fedeca.fedeca_core.FedECA`.
+    return_propensities: bool
+        return propensity scores in the results dataframe, by default True
+    return_weights: bool
+        return samples weights in the results dataframe, by default True
+
+    Returns
+    -------
+    pd.DataFrame
+        Results of the experiment of `n` rows where `n` is the number of models.
+    """
+    if seed is None:
+        seed = data_gen.rng
+    if fit_fedeca is None:
+        fit_fedeca = {}
+    rng = np.random.default_rng(seed)
+
+    models = dict(
+        (name, hydra.utils.instantiate(model)) for name, model in model_configs.items()
+    )
+    for model in models.values():
+        model.set_random_state(rng)
```

Comment:
Nice ! Can you add tests by using old versions of repo (prior to this branch) to store results of run_experiment and compare new pipeline with old one ?
I can do in a few weeks it if you don't have time.